graph/coloring/graph_test.go

```diff
@@ -0,0 +1,150 @@
+package coloring_test
+
+import (
+	"github.com/TheAlgorithms/Go/graph/coloring"
+	"strconv"
+	"testing"
+)
+
+type testGraph struct {
+	Graph        *coloring.Graph
+	ColorsUsed   int
+	VertexColors map[int]coloring.Color
+}
+
+func getTestGraphs() (list []*testGraph) {
+	// Graph 0th:
+	// 		1---2
+	// 		| /  \
+	//  4---3     0
+	// Min number of colors required = 3
+	g0 := &testGraph{
+		Graph:      &coloring.Graph{},
+		ColorsUsed: 3,
+		VertexColors: map[int]coloring.Color{
+			1: 1, 4: 1, 0: 1,
+			2: 2,
+			3: 3,
+		},
+	}
+	list = append(list, g0)
+	g0.Graph.AddEdge(4, 3)
+	g0.Graph.AddEdge(3, 1)
+	g0.Graph.AddEdge(3, 2)
+	g0.Graph.AddEdge(1, 2)
+	g0.Graph.AddEdge(2, 0)
+
+	// Graph 1st:
+	// 		1---2
+	// 		| / |
+	//  4---3---0
+	// Min number of colors required = 3
+	g1 := &testGraph{
+		Graph:      &coloring.Graph{},
+		ColorsUsed: 3,
+		VertexColors: map[int]coloring.Color{
+			1: 1, 4: 1, 0: 1,
+			2: 2,
+			3: 3,
+		},
+	}
+	list = append(list, g1)
+	g1.Graph.AddEdge(4, 3)
+	g1.Graph.AddEdge(3, 1)
+	g1.Graph.AddEdge(3, 2)
+	g1.Graph.AddEdge(1, 2)
+	g1.Graph.AddEdge(2, 0)
+	g1.Graph.AddEdge(3, 0)
+
+	// Graph 2nd:
+	// 		1---2
+	// 		|
+	//  4---3   0
+	// Min number of colors required = 2
+	g2 := &testGraph{
+		Graph:      &coloring.Graph{},
+		ColorsUsed: 2,
+		VertexColors: map[int]coloring.Color{
+			1: 1, 4: 1, 0: 1,
+			2: 2, 3: 2,
+		},
+	}
+	list = append(list, g2)
+	g2.Graph.AddVertex(0)
+	g2.Graph.AddEdge(4, 3)
+	g2.Graph.AddEdge(3, 1)
+	g2.Graph.AddEdge(1, 2)
+
+	// Graph 3rd:
+	// 	1---2	4
+	// 	|	|	|
+	//  0---3	5
+	// Min number of colors required = 2
+	g3 := &testGraph{
+		Graph:      &coloring.Graph{},
+		ColorsUsed: 2,
+		VertexColors: map[int]coloring.Color{
+			1: 1, 3: 1, 4: 1,
+			0: 2, 2: 2, 5: 2,
+		},
+	}
+	list = append(list, g3)
+	g3.Graph.AddEdge(0, 3)
+	g3.Graph.AddEdge(1, 2)
+	g3.Graph.AddEdge(1, 0)
+	g3.Graph.AddEdge(3, 2)
+	g3.Graph.AddEdge(4, 5)
+
+	// Graph 4th:
+	// Completely Connected graph of vertex 4
+	// Min number of colors required = 2
+	g4 := &testGraph{
+		Graph:      &coloring.Graph{},
+		ColorsUsed: 4,
+		VertexColors: map[int]coloring.Color{
+			0: 1, 1: 2, 2: 3, 4: 4,
+		},
+	}
+	list = append(list, g4)
+	for i := 0; i < 4; i++ {
+		for j := i + 1; j < 4; j++ {
+			g4.Graph.AddEdge(i, j)
+		}
+	}
+
+	return
+}
+
+func TestGraph_ValidateColorsOfVertex(t *testing.T) {
```

Comment:
I don't like the underscores being used in Go. It kind of distracts me 😅
```suggestion
func TestGraphValidateColorsOfVertex(t *testing.T) {
```
But there is no hard naming conventions on tests in the Go community so it would be fine even if you leave it. I would say that I don't think your names would clash with other implementations so it is worth considering my suggestion above. But again no hard rule on this 😄

Reply:
> I don't like the underscores being used in Go. It kind of distracts me 😅

I also, don't like them, but the Go's test function generator keeps on using it

Reply:
Which one are you referring to? I haven't seen a test generator supported by gotools - I'd like to know! 😃

Reply:
Ah, no-no, I use GoLand IDE and it provides shortcuts for creating test functions. 
Even I haven't seen any test generator by gotools, I think it uses https://github.com/cweill/gotests, not sure.

Reply:
Ah cool, then I think you should remove the underscore since the whole repo is using underscores. Thanks!